### PR TITLE
feat: support iceberg_v2 sink writer

### PR DESCRIPTION
**Pull request by ZENOTME**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

support append only writer for now

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- ~~[ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).~~
- ~~[ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).~~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- ~~[ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)~~

- ~~[ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))~~

## Documentation

- [x] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/connector/src/sink/iceberg.rs [hunk 1]
```diff
@@ -100,7 +104,50 @@ impl Debug for IcebergSink {
...
+            return Err(SinkError::Iceberg(format!(
+                "Schema not match, expect: {:?}, actual: {:?}",
+                self.schema, iceberg_schema
+            )));
+        }
+
+        Ok(table)
+    }
+
+    fn parse_bucket_and_root_from_path(config: &IcebergConfig) -> Result<(String, String)> {
```
Comment (liurenjie1024) [inline src/connector/src/sink/iceberg.rs:149]: Is it possible to use `url` crate?

File: src/connector/src/sink/iceberg.rs [hunk 2]
```diff
@@ -166,38 +199,113 @@ impl Sink for IcebergSink {
...
+            .map_err(|err| SinkError::Iceberg(err.to_string()))?;
+
+        Ok(())
     }
 
     /// Clean up
     async fn abort(&mut self) -> Result<()> {
-        todo!()
+        return Err(SinkError::Iceberg(
+            "Iceberg sink can't support abort now".to_string(),
```
Comment (liurenjie1024) [inline src/connector/src/sink/iceberg.rs:302]: In `abort` we should delete finished but uncommitted files. We can postpone this in later pr, but we should not return the error here. We should just ignore it.

**Review by liurenjie1024 — APPROVED** [review]
LGTM

File: src/connector/src/sink/iceberg.rs [hunk 3]
```diff
@@ -146,10 +130,71 @@ impl IcebergSink {
...
+            format!("{}/{}/{}", root, config.database_name, config.table_name)
+        };
+
+        Ok((bucket.to_string(), table_root_path))
+    }
+
+    pub fn new(config: IcebergConfig, schema: Schema) -> Result<Self> {
+        let (bucket_name, table_root) = Self::parse_bucket_and_root_from_path(&config)?;
+        // TODO(ZENOTME): Only support append-only mode now.
+        if !config.force_append_only {
```
Comment (tabVersion) [inline src/connector/src/sink/iceberg.rs:187]: it is possible the upstream is append only. better also check sinktype

**Review by tabVersion — APPROVED** [review]

**Pull request merged**